twisted/test/proto_helpers.py

```diff
@@ -695,3 +702,86 @@ def connectUNIX(self, address, factory, timeout=30, checkPID=0):
         Fake L{IReactorUNIX.connectUNIX}, that raises L{_connectException}.
         """
         raise self._connectException
+
+
+
+@implementer(ILogFilterPredicate)
+class NamespaceFilterPredicate(object):
```

Comment:
This is so close to the job that [`twisted.logger.LogLevelFilterPredicate`](https://twistedmatrix.com/documents/16.3.0/api/twisted.logger.LogLevelFilterPredicate.html) does that it makes me wonder if this is duplication.  The reason that `LogLevelFilterPredicate` can't simply do this out of the box is we have no "a level so high you should never log it" level.

Perhaps we could add that as a very small PR first - add a way to express to `LogLevelFilterPredicate` "just don't log anything in this namespace", which could be passed to the `defaultLogLevel` parameter to `__init__` ?  I'd rather avoid having a proliferation of lots of tiny filter predicates which do _almost_ the same thing.